components/browser/storage-sync/src/main/java/mozilla/components/browser/storage/sync/PlacesStorage.kt

```diff
@@ -67,15 +69,50 @@ abstract class PlacesStorage(
         }
     }
 
-    /**
-     * Cleans up background work and database connections
-     */
+    @Deprecated(
+        "Use `cleanupWrites` and `cleanupReads` to get a similar functionality. " +
+            "See https://github.com/mozilla-mobile/android-components/issues/7348 for a description of the issue."
+    )
     override fun cleanup() {
         writeScope.coroutineContext.cancelChildren()
         readScope.coroutineContext.cancelChildren()
         places.close()
```

Comment:
Looked into removing this `places.close()` call and delegate to the new `cleanupWrites` and `cleanupReads` methods but saw some tests failing in the `mc.browser.storage.sync` package.

Reply:
OK, no reason to remove it as part of this work. I just pointed out because I remembered why this method wasn't used.